packages/ui-kit/src/components/PieChart/PieChart.tsx

```diff
@@ -0,0 +1,341 @@
+          setPropsMismatch(true)
+          return
+        }
+
+        if (isStatic && (!headers || !rows)) {
+          // console.error('InvalidPropsError: When passing the data via props you must pass both `headers` and `rows`') we will set logs as a feature later
+          setPropsMismatch(true)
+
+          return
+        }
+
+        if (
+          !isStatic &&
+          (hasError?.name === 'AccessTokenError' ||
+            !query.metric ||
+            !query.timeRange ||
+            !query.dimensions ||
+            !query.rowLimit)
+        ) {
+          // console.error(
+          //   'InvalidPropsError: When opting for fetching data you must pass at least `accessToken`, `metric`, `dimensions`, `rowLimit` and `timeRange` in the `query` prop'
+          // ) we will set logs as a feature later
+          setPropsMismatch(true)
+          return
+        }
+
+        if (variant !== 'pie' && variant !== 'doughnut') {
+          // console.error('InvalidPropsError: `variant` prop must be either `pie` or `doughnut`') we will set logs as a feature later
+          setPropsMismatch(false)
+        }
+
+        setPropsMismatch(false)
+      }
+
+      if (!isLoadingStatic) {
+        handlePropsMismatch()
+      }
+    }, [isStatic, headers, rows, query, isLoadingStatic, variant, hasError?.name])
+
+    React.useEffect(() => {
+      if (isStatic) {
+        renderChart({ headers, rows })
+      }
+    }, [isStatic, isLoadingStatic, variant, headers, rows, renderChart])
+
+    React.useEffect(() => {
+      if (fetchedData?.leaderboard && !isStatic) {
+        renderChart(fetchedData.leaderboard)
+      }
+    }, [fetchedData, variant, isStatic, renderChart])
+
+    React.useEffect(() => {
+      return () => {
+        destroyChart()
+      }
+    }, [])
+
+    if (hasError || propsMismatch) {
+      destroyChart()
+      return <ErrorFallback {...errorFallbackProps} error={error} />
+    }
+
+    const isNoContainerRef = (variant === 'pie' || variant === 'doughnut') && !canvasRef.current
+
+    if (((isStatic && isLoadingStatic) || (!isStatic && isLoading)) && isNoContainerRef) {
+      destroyChart()
+      return <Loader {...loaderProps} />
+    }
+
+    const totalValue = isStatic ? rows?.reduce((a, b) => a + Number(b[1]), 0) ?? 0 : Number(counterData?.counter?.value)
+
+    const getListItem = () => {
+      if (showValues) {
+        const _rows = isStatic ? rows : fetchedData?.leaderboard?.rows
+
+        return (
+          <div className={classnames(componentStyles.pieChartList, className)}>
+            <ul>
+              {_rows?.map((row, index) => (
+                <li key={`label-${index}`}>
+                  <span>{row[0]}</span>
+                  <span>{parseFloat(row[1] ?? '0').toLocaleString()}</span>
+                </li>
+              ))}
+            </ul>
+          </div>
+        )
+      }
+    }
+
+    return (
+      <>
```

Comment:
We should put `className` prop only once and on the component's top-level element. It will allow the customers to style the top element itself as well as any child element. For instance [packages/ui-kit/src/components/Leaderboard/Leaderboard.tsx](https://github.com/propeldata/ui-kit/blob/main/packages/ui-kit/src/components/Leaderboard/Leaderboard.tsx#L382).
The same logic applies to `ref`.

Currently `className` is spread across multiple elements. So, if define a custom class with `border: 1px solid #ff0000` and pass it via `className` prop, I will get:
<img width="960" alt="image" src="https://github.com/propeldata/ui-kit/assets/55801864/b90472b8-97f1-41d6-9ee7-85ec69d0d3e2">

Reply:
yea, that is true. I missed it. I fix it not.